web/src/settings_org.js

```diff
@@ -700,6 +700,65 @@ export function set_up() {
     maybe_disable_widgets();
 }
 
+function set_up_dropdown_widget(setting_name, setting_options, setting_type) {
+    const $save_discard_widget_container = $(`#id_${CSS.escape(setting_name)}`).closest(
+        ".settings-subsection-parent",
+    );
+    const $events_container = $(`#id_${CSS.escape(setting_name)}`).closest(".settings-section");
+
+    let text_if_current_value_not_in_options;
+    if (setting_type === "channel") {
+        text_if_current_value_not_in_options = $t({defaultMessage: "Cannot view channel"});
+    }
+
+    let unique_id_type = dropdown_widget.DataTypes.NUMBER;
+    if (setting_type === "language") {
+        unique_id_type = dropdown_widget.DataTypes.STRING;
+    }
+
+    const setting_dropdown_widget = new dropdown_widget.DropdownWidget({
+        widget_name: setting_name,
+        get_options: setting_options,
+        $events_container,
+        item_click_callback(event, dropdown) {
+            dropdown.hide();
+            event.preventDefault();
+            event.stopPropagation();
+            const dropdown_widget =
+                settings_components.get_widget_for_dropdown_list_settings(setting_name);
+            dropdown_widget.render();
+            settings_components.save_discard_realm_settings_widget_status_handler(
+                $save_discard_widget_container,
+            );
+        },
+        tippy_props: {
+            placement: "bottom-start",
+        },
+        default_id: realm[setting_name],
+        unique_id_type,
+        text_if_current_value_not_in_options,
+        on_mount_callback(dropdown) {
+            if (setting_type === "group") {
+                $(dropdown.popper).css("min-width", "300px");
+            }
+        },
+    });
+    settings_components.set_dropdown_setting_widget(setting_name, setting_dropdown_widget);
+    setting_dropdown_widget.setup();
+}
+
+export function set_up_dropdown_widget_for_realm_group_settings() {
+    const realm_group_permission_settings = Object.keys(
+        realm.server_supported_permission_settings.realm,
```

Comment:
This would mean that we should add the UI element in the commit which adds the setting in backend. Or else we can just call the functions for each setting without using a loop or define a list of settings seprately and then use for loop.

Reply:
This might be OK; I think we expect once this work has progressed for the frontend changes to be quite small. It's worth posting about in #frontend for visibility.